### PR TITLE
PhysicalFileProvider: Remove second try on GetFileLinkTargetLastWriteTimeUtc 

### PR DESCRIPTION
**Pull request by adamsitnik**
https://github.com/dotnet/runtime/pull/56915/files#r686113979

### CODE REVIEW

**Review by jeffhandley — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.FileProviders.Physical/src/Internal/FileSystemInfoHelper.cs [hunk 1]
```diff
@@ -65,7 +65,7 @@ public static bool IsExcluded(FileSystemInfo fileSystemInfo, ExclusionFilters fi
                     // Try one more time, if it fails again just give up.
                     if (!isSecondTry)
                     {
-                        GetFileLinkTargetLastWriteTimeUtc(fileInfo, isSecondTry: true);
+                        return GetFileLinkTargetLastWriteTimeUtc(fileInfo, isSecondTry: true);
```
[the remark this answers is not in the document]
Comment (danmoseley) [reply src/libraries/Microsoft.Extensions.FileProviders.Physical/src/Internal/FileSystemInfoHelper.cs:68]: > PhysicalFileProvider code was already kinda resilient to these errors when used with non-link files so maybe we should go for resiliency instead?

So we have made PFP potentially less resilient for existing code? On the face of it that sounds like something we should avoid but I don't have full context.

[the remark this answers is not in the document]
Comment (Jozkee) [reply src/libraries/Microsoft.Extensions.FileProviders.Physical/src/Internal/FileSystemInfoHelper.cs:68]: > So we have made PFP potentially less resilient for existing code?

Actually, nevermind that last thing, I was thinking that `File.GetLastWriteTimeUtc` and `FileSystmeInfo.LastWriteTimeUtc` was more resilient to exceptions but they are not.

File: src/libraries/Microsoft.Extensions.FileProviders.Physical/src/Internal/FileSystemInfoHelper.cs [hunk 2]
```diff
@@ -63,10 +63,6 @@ public static bool IsExcluded(FileSystemInfo fileSystemInfo, ExclusionFilters fi
                 {
                     // The file ceased to exist between LinkTarget and ResolveLinkTarget.
                     // Try one more time, if it fails again just give up.
```
Comment (danmoseley) [inline src/libraries/Microsoft.Extensions.FileProviders.Physical/src/Internal/FileSystemInfoHelper.cs:65]:
```suggestion
```

**Review by danmoseley — APPROVED** [review]

**Comment by danmoseley** [comment]
Thanks, I think this is cleaner.

**Comment by danmoseley** [comment]
@tarekgh seems several test failures related to your earlier work? https://github.com/dotnet/runtime/pull/57136/checks?check_run_id=3294583322

**Comment by Jozkee** [comment]
I just filed an issue for `Microsoft.Extensions.Hosting.Tests.ConsoleLifetimeExitTests` https://github.com/dotnet/runtime/issues/57184. 
@danmoseley seems to me that all the CI failures are unrelated. Will merge now.

**Pull request merged**